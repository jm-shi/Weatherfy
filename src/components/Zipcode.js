import React from 'react';
import Searchbar from './Searchbar';

class Zipcode extends React.Component {
  render() {
    return (
      <div>
        <form className="form">
          <div className="form-container">
            <label id="search-label">Get weather by zipcode</label>
            <Searchbar />
          </div>
        </form>
      </div>
    );
  }
}

module.exports = Zipcode;
