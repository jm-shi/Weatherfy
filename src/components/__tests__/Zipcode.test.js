import React from 'react';
import Zipcode from '../Zipcode.js';
import { shallow } from 'enzyme';

test('should render Zipcode correctly', () => {
  const wrapper = shallow(<Zipcode />);
  expect(wrapper).toMatchSnapshot();
});
