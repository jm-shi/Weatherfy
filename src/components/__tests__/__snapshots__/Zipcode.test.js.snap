// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Zipcode correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Zipcode />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="form"
      >
        <div
          className="form-container"
        >
          <label
            id="search-label"
          >
            Get weather by zipcode
          </label>
          <Searchbar />
        </div>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-container"
        >
          <label
            id="search-label"
          >
            Get weather by zipcode
          </label>
          <Searchbar />
        </div>,
        "className": "form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              id="search-label"
            >
              Get weather by zipcode
            </label>,
            <Searchbar />,
          ],
          "className": "form-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Get weather by zipcode",
              "id": "search-label",
            },
            "ref": null,
            "rendered": "Get weather by zipcode",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="form"
        >
          <div
            className="form-container"
          >
            <label
              id="search-label"
            >
              Get weather by zipcode
            </label>
            <Searchbar />
          </div>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="form-container"
          >
            <label
              id="search-label"
            >
              Get weather by zipcode
            </label>
            <Searchbar />
          </div>,
          "className": "form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                id="search-label"
              >
                Get weather by zipcode
              </label>,
              <Searchbar />,
            ],
            "className": "form-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Get weather by zipcode",
                "id": "search-label",
              },
              "ref": null,
              "rendered": "Get weather by zipcode",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
